### ISSUE TITLE
ECT being registered already has an induction completion date

### ISSUE THREAD

**Issue by mason-emily** [DFE-Digital/register-early-career-teachers#687]
## Description
This story delivers the functionality to check for an induction completion date associated with the ECT's TRS record when we find a match. ECTs who already have an induction completion date should not be registered in the service as they have already completed induction, so we end the registration journey in this case.

## User story

**As** a school user
**I need** to be told if the ECT I'm registering has already completed induction
**So that** I know why I cannot register the ECT

## Prototype, screenshots & content
[Prototype link to the 'You cannot register [ECT name]' page](https://ecf2-prototy-v4-k7mv50fp1jl0ue.herokuapp.com/v4/other-pages/add-ect/induction-completed)

Screenshots of relevant screens for if the ECT ****does have an induction completion date** on their TRS record:
![image](https://github.com/user-attachments/assets/8d0904bd-46db-4867-bb11-5ec2eb77a392)

Content should be as per the prototype.
URL should be as per [this doc](https://educationgovuk.sharepoint.com/:x:/r/sites/TeacherServices/Shared%20Documents/Teacher%20Continuing%20Professional%20Development/Teacher%20CPD%20Team/Register%20early%20career%20teachers/Beta/Design/Register%20ECTs%20page%20titles%20and%20URLs.xlsx?d=w0c1578ea643147b29357c7f610ee178f&csf=1&web=1&e=U8iIdi).

## Acceptance criteria

- [x] 1. When user submits a valid TRN and date of birth on the <ins>'Find an ECT'</ins> page and a match is found for the ECT's TRS record using both TRN and date of birth:
    - [x] 1a. If the ECT's record **has** an induction completion date, user will be navigated to the <ins>'You cannot register [ECT name]'</ins> page. 
    - [x] 1b. If the ECT's record **does not have** an induction completion date, user will continue to be navigated to the <ins>'Check if this is your ECT'</ins> page.
- [x] 2. On the <ins>'You cannot register [ECT name]'</ins> page:
    - [x] 2a. H1 text will include the ECT's first and last name from their TRS record 
    - [x] 2b. Content will be as per the prototype - note, this is slightly different to #688 .
    - [x] 2c. There will be a link to 'Register another ECT' which navigates the user to the <ins>'Find an ECT'<ins> page where the fields should be reset to blank.
    - [x] 2d. There will be no 'Back' link on the <ins>'You cannot register [ECT name]'</ins> page.
- [x] 3. AC 1&2 above will be replicated for the <ins>'We cannot find your ECT's details'</ins> page when a TRS match is found using TRN and NINO.

## Technical notes
Note, the induction completion date to check is the **overall** induction completion date i.e. not an induction period date. @tonyheadford would it be possible to confirm the exact name?

## Linked tickets
Dependent on #512 
Very similar to #688 
Linked to #473 #689 #688

**Comment by claire-hughez** [comment]
TRS seed data for us to use:

2600157, 02/03/1976 is Exempt
2600150, 07/10/1974 is Passed (with a completion date)

**Comment by gracecbrennan** [comment]
@TobyRet is going to park this and come back to it

**Comment by TobyRet** [comment]
@claire-hughez  

`2500150` does not appear to have induction data?

This is the response from TRS API

```
{"trn"=>"2600150",
 "firstName"=>"Becki",
 "middleName"=>"",
 "lastName"=>"Brown",
 "dateOfBirth"=>"1974-10-07",
 "nationalInsuranceNumber"=>nil,
 "emailAddress"=>"test@test.net",
 "qts"=>{"awarded"=>"2019-08-01", "certificateUrl"=>"/v3/certificates/qts", "statusDescription"=>"Qualified"},
 "eyts"=>nil}
```

**Comment by gracecbrennan** [comment]
Claire is going to follow up with James Gunn on this

**Comment by claire-hughez** [comment]
Confirmed it's because we need to have a parameter to expose induction data, it's not done by default.

**Comment by gracecbrennan** [comment]
Toby will jump back on this and will also wait for refactoring work to be finished.

**Comment by gracecbrennan** [comment]
As above - dependency on refactoring work

**Comment by TobyRet** [comment]
@claire-hughez - adding `?include=Induction` to the request params is bringing back the induction info ...

**Comment by claire-hughez** [comment]
thanks for confirming

**Comment by mason-emily** [comment]
This looks great! 

Would it be possible to add the national insurance number for the induction completed record TRN 2600150 so I can test AC3?

**Comment by mason-emily** [comment]
Oh I think the check is based on TRN only so you can never get to the NiNo page if the TRN has an induction completion date - this might be fine, or is a privacy concern to not match on 2 data points before playing back the name? @claire-hughez

**Comment by claire-hughez** [comment]
I think we need to match on 2 datapoints. I know it seems silly, but if someone was to put in a random TRN in there, they could find out about the state of their induction or extra info. I think it's best to stick to this for any kind of checks.

**Comment by TobyRet** [comment]
@mason-emily @claire-hughez 

The test user for the 'induction completed' journey does not not have an ECT. 

This is what we get back from the TRS API ...

```
{"induction"=>
  {"startDate"=>"2020-09-01",
   "endDate"=>"2021-03-01",
   "status"=>"Pass",
   "statusDescription"=>"Pass",
   "certificateUrl"=>"/v3/certificates/induction"},
 "trn"=>"2600150",
 "firstName"=>"Becki",
 "middleName"=>"",
 "lastName"=>"Brown",
 "dateOfBirth"=>"1974-10-07",
 "nationalInsuranceNumber"=>nil,
 "emailAddress"=>"test@test.net",
 "qts"=>{"awarded"=>"2019-08-01", "certificateUrl"=>"/v3/certificates/qts", "statusDescription"=>"Qualified"},
 "eyts"=>nil}
```

**Comment by claire-hughez** [comment]
Hi Toby, it's not that they have to match on National insurance number, it's just that the check where we pull back that their induction is complete needs to work not just on TRN, but on TRN + one other piece of personal info. So in this case, date of birth. Is that the case in the code, or not?

**Comment by mason-emily** [comment]
We do also need to be able to test with national insurance number - is there another record we could try in that case?

**Comment by claire-hughez** [comment]
_We do also need to be able to test with national insurance number - is there another record we could try in that case?_

I can ask @gunndabad and Rob.

**Comment by TobyRet** [comment]
I've pushed a fix - the user is redirect to the 'induction completed' page if the teacher has completed their induction AND the date of birth matches

**Comment by mason-emily** [comment]
Looks like it's working well to me!

We'll need the national insurance to do the last check that that works at some point

**Comment by claire-hughez** [comment]
Thanks Toby. I have pinged TRA folks, but not heard back yet. I don't want to keep this PR hanging for too long. Should we merge and then test with the DOB and TRN they recommend later? What do you think Toby and Emily?

**Comment by TobyRet** [comment]
@claire-hughez - for info: a national insurance number is being returned for the 'exempt' user (just need one for the 'induction completed' user)

**Comment by claire-hughez** [comment]
NINO and completion date:
1000890
JL049125A
Adam Aparicio
15/01/1997

NINO and exempt
3002564
JK028223B
Effie Abernathy
05/07/1991

**Comment by claire-hughez** [comment]
Toby is updating the content in another ticket. All the other AC is met, so this is good to move to done.
